app/helpers/generic_file_helper.rb

```diff
@@ -39,7 +39,19 @@ def download_path(*args)
       gf = GenericFile.find(item.id)
       return gf.doi_url if gf.doi_url.present?
     end
-    return sufia.download_path(*args)
+    if args[1].nil?
+      return "/files/" + item.id + "/" + URI::encode(item.label) unless item.label.nil?
+      return "/files/" + item.id + "/nofile"
+    else
+      path = "/files/" + item.id + "?"
+      i = 0
+      args[1].each do |key,value|
```

Comment:
What's the significance of args[1] here? None of the use cases listed here involve a collection being the second argument: https://github.com/ualbertalib/HydraNorth/blob/2c1f881dd14aad385eb2760cb8c5de21c794718a/app/helpers/generic_file_helper.rb#L26-L31

It would be really good to document what this thing is expecting args[1] to be for this case. I also don't see any guard logic to ensure that args[1] *is* a collection. This will break several of the call-types documented in that comment will break, eg) download_path(@generic_file, file: 'webm')